minimesos/src/main/java/com/containersol/minimesos/main/CommandInstall.java

```diff
@@ -72,7 +73,11 @@ public void execute() throws MinimesosException {
 
         MesosCluster cluster = ClusterRepository.loadCluster(new MesosClusterContainersFactory());
         if (cluster != null) {
-            cluster.install(marathonJson);
+            Marathon marathon = cluster.getMarathon();
+            if (marathon == null) {
+                throw new MinimesosException("Marathon container is not found in cluster " + cluster.getClusterId());
+            }
+            marathon.deployApp(marathonJson);
         } else {
             throw new MinimesosException("Running cluster is not found");
         }
```

Comment:
Reorder the statements. Check first if cluster is running and has Marathon, then look into JSON